src/device/device-management.ts

```diff
@@ -33,29 +33,87 @@ export type AudioDeviceConstraints = {
   deviceId?: string;
 };
 
-export type VideoDeviceConstraints = {
-  deviceId?: ConstrainDOMString;
-  width?: ConstrainULong;
-  height?: ConstrainULong;
-  aspectRatio?: ConstrainDouble;
-  frameRate?: ConstrainDouble;
-  facingMode?: ConstrainDOMString;
+// Predefined encoder config mentioned similar to https://api-ref.agora.io/en/voice-sdk/web/4.x/globals.html#videoencoderconfigurationpreset
+export enum StaticVideoEncoderConfig {
```

Comment:
What is this used for? Wondering if it makes sense for this to be defined in core.

Reply:
Even if we do have this, instead of being it's own type, we should just define named, constant instances of `VideoEncoderConfig` that have the fields pre-set.

Reply:
Will clean up the comments but the staticVideoEncoderConfig is a enum the developers would have to do 

constrains.encoderConfig.preDefinedEncoderConfig compared to constraints.encoderConfig 

is that the ask !?

Reply:
No, what I'm saying is this:

rather than having two types, e.g.:
```
export type VideoEncoderConfig {
   maxHeight: number,
   maxWidth: number,
   maxFps: number,
   // etc (just an example)
}
export enum StaticVideoEncoderConfig {
    1080p,
    720p,
    360p
}
```
do:
```
export type VideoEncoderConfig {
   maxHeight: number,
   maxWidth: number,
   maxFps: number,
   // etc (just an example)
}

export const highQualityVideoEncoiderConfig: VideoEncoderConfig = {
   maxHeight: 1080,
   maxWidth: 1920,
   maxFps: 30,
}
```

So: a single type that is always used, and pre-defined settings are just defined as constant values of that type, so that way the user can have convenience if it wants, and the code only has to deal with a single type.